### ISSUE TITLE
Nvidia freestyle filters not working (Ray tracing and SSAO)

### ISSUE THREAD

**Issue by mathewriver** [yuzu-emu/yuzu#10367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

1237

### Description of Issue

By changing the name of the executable to ACU.exe (Assassin's Creed Unity) and adding it to Nvidia control panel, setting DXGI I've managed to get Nvidia Freestyle filters to work with Yuzu, except RTGI (Ray tracing) and SSAO. While in game it says that the "SSRTGI" filter uses depth information that is not available in this application or scene, same for SSAO, but I can use every other effect except RTGI and SSAO.

### Expected Behavior

Be able to use RTGI and SSAO from Nvidia Freestyle filters.

### Reproduction Steps

Step 1: Rename Yuzu executable to ACU.exe.

Step 2: Right click on your desktop area, open Nvidia control panel and then go to 3D settings > Manage 3D settings > Program Settings, mark the option Show only programs found on this computer and then add ACU.exe (Remember, this is the Yuzu executable) to the programs list, move to the program settings list, search for Vulkan/OpenGL present method, open it and chose Prefer Layered on DXGI Swapchain, apply settings.

Step 3: Open Yuzu emulator and change the API to Vulkan.

Step 4: You should be good to go now. Open Cemu, open Zelda BOTW, wait until the shaders are compiled, then see if the overlay will show up on the right top side of the window, if it does, press either Alt + Z or Alt + F3 to get started, then set the effects as you wish (same as in the Cemu method).
More details here: https://lucrebem.com.br/en/gaming/cemu-yuzu-raytracing-nvidia

### Log File

There's no log file.

### System Configuration

CPU: intel core i5
GPU/Driver: RTX 2060
RAM: 8Gb
OS: Windows 10

**Comment by mathewriver** [comment]
So you're saying that Depth of Buffer was never supported by Yuzu?

**Comment by mathewriver** [comment]
So you're saying that Depth Buffer was never supported by Yuzu?

**Comment by liamwhite** [comment]
What does freestyle need the graphics application to do in order for this to work? We could evaluate the feasibility based on that information.

**Comment by mathewriver** [comment]
Nvidia Freestyle has "SSRTGI" (Ray Tracing) and SSAO (Screen Space Ambient Occlusion) filters and both of them need depth information to work, but that depth information is not available when you're playing a game in Yuzu. So basically those 2 filters are requiring z-buffering to apply the post-processing effects. Even when you use Reshade (which is not what we're discussing here, but it's an alternative to Nvidia Freestyle) you end up having problems with depth buffer in Yuzu, like a flickering screen, so it propably has something to with the emulator, not the software trying to apply the effects. E.g. when you see Zelda Breath of the Wild in Cemu emulator, in the graphics pack there's an option to prepare the game for Reshade (certainly to prepare it for depth buffer), there's nothing like that in Yuzu, but still I suppose this shouldn't be a feature request, since Switch games do have z-buffering, the problem is: an external software (like Nvidia) is not having access that information or (like Reshade) not having proper access to that information in order to apply the post-processing effects.

**Comment by liamwhite** [comment]
Right, but where in the pipeline does it need the depth buffer to be used?

**Comment by mathewriver** [comment]
I'm sorry, I'm not a developer to know how to give you that information, I'm just reporting a problem with the software.

**Comment by goldenx86** [comment]
This requires mod support per game to be implemented, and we don't do mods, the community does.
If a mod for reshade is made, this could be implemented, but right now, it's not a priority, we're focused on bugfixing, compatibility and performance first.

**Comment by matt7570** [comment]
The mod support per game was just an example from Cemu. Can't it be globally implemented in Yuzu?
Even Nvidia control panel says that Yuzu doesn't support Ambient Occlusion and other realistic effects:

[url=https://ibb.co/nj0TpZ0][img]https://i.ibb.co/3cswKGs/image.png[/img][/url]

**Comment by matt7570** [comment]
The mod support per game was just an example from Cemu. Can't it be globally implemented in Yuzu?
Even Nvidia control panel says that Yuzu doesn't support Ambient Occlusion and other realistic effects:

<a href="https://ibb.co/nj0TpZ0"><img src="https://i.ibb.co/3cswKGs/image.png" alt="image" border="0"></a>

**Comment by matt7570** [comment]
The mod support per game was just an example from Cemu. Can't it be globally implemented in Yuzu?
Even Nvidia control panel says that Yuzu doesn't support Ambient Occlusion and other realistic effects:

<a href="https://ibb.co/GfZL2YP"><img src="https://i.ibb.co/CpjZ7C0/image.png" alt="image" border="0"></a>

**Comment by liamwhite** [comment]
I know that it says that, but I don't know why - I need to know what it's looking for in order to implement any support for these.

**Comment by matt7570** [comment]
@liamwhite Well I'm willing to help. What I can say is that I'm trying to apply shadows and ray traced light effects to Yuzu games, but Nvidia Freestyle says it's not receiving any depth buffer information from Yuzu. But with Reshade I am able to achieve something, just that it's buggy, the graphics will keep flickering. This video is from Cemu, but it uses Reshade, this is what I'm trying to achieve with Yuzu (with either Nvidia Freestyle or Reshade):

https://www.youtube.com/watch?v=i8mFHzjQq_w

**Comment by matt7570** [comment]
@liamwhite Sorry if I can't be of much help here, but I could go around and ask other people about the technical details and then provide you with feedback, if I find something, can I bring the info for you?

**Comment by matt7570** [comment]
I found out how to enable the log option, but there are too many options, which ones should I choose in order to provide a log file here?

**Comment by matt7570** [comment]
Can anyone reopen this issue?
